types/index.d.ts

```diff
@@ -0,0 +1,153 @@
+import type { Client } from 'aedes';
+import type { AedesPacket } from 'aedes-packet';
+import type { Readable } from 'stream';
+
+export type { AedesPacket as Packet } from 'aedes-packet';
+
+export interface AedesPersistenceSubscription {
+  clientId?: string;
+  topic: string;
+  qos: number;
+}
```

Comment:
This seems not correct. AFAIK subscriptions are a map `clientId : { topic: qos }`

Ref: https://github.com/moscajs/aedes-persistence/blob/master/persistence.js#L74-L103

Reply:
In fact this interface is probably correct, it matches the argument `subs` received by `addSubscriptions`, `removeSubscriptions`...
But the `_subscriptions` are not correct like you noticed, it's a Map<string, Map<string, string>>

Reply:
In that case why don't use aedes sub interface? btw my comment was more about the use of this in `_subscriptions`

Reply:
The initial reason is that i wanted to avoid importing Aedes in that project. Now when i'm looking at the `Subscription` interface in aedes (which leads to `ISubscription` from mqtt-packet), i don't see the `clientId` property.

Reply:
The client id used if from the client property, then the subscription added to the trie is in the format you defined in this interface. But the subscriptions array is retrived from the mqtt packet:

https://github.com/moscajs/aedes/blob/master/lib/handlers/subscribe.js#L108